lib/arux_app/api/auth.rb

```diff
@@ -60,6 +60,32 @@ def authorization_url(scope: "public")
         base_uri.to_s
       end
 
+      def basic_authentication(username, password, scope = "public")
+        params = {
+          scope: scope,
+          grant_type: "password",
+          client_id: client_id,
+          client_secret: client_secret
+        }
+        request = HTTPI::Request.new
+        request.url = "#{self.class.server_uri}/oauth/token"
+        request.body = params
+        request.headers = { 'User-Agent' => USER_AGENT }
+        request.auth.basic(username, password)
+        response = HTTPI.post(request)
+
+        if !response.error?
+          AccessToken.new(
+            token: JSON.parse(response.body)['access_token'],
+            scope: JSON.parse(response.body)['scope'],
+            auth: self
+          )
+        else
+          raise(API::Error.new(response.code, response.body))
+        end
```

Comment:
You could do a guard here:

```suggestion
        raise(API::Error.new(response.code, response.body)) if response.error?
        
        AccessToken.new(
          token: JSON.parse(response.body)['access_token'],
          scope: JSON.parse(response.body)['scope'],
          auth: self
        )
```

Reply:
Nice.